packages/website/src/docs/language-basics/namespaces.md

```diff
@@ -0,0 +1,98 @@
+---
+id: namespaces
+title: Namespaces
+---
+
+# Namespaces
+
+Namespaces let you group related types together into namespaces. This helps organize your types, making them easier to find and prevents name conflicts. Namespaces are merged across files, so you can reference any type anywhere in your Cadl program via its namespace.
+
+## Basics
+
+Create a namespace with the `namespace` keyword.
+
+```cadl
+namespace SampleNamespace {
+  model SampleModel {}
+}
+```
+
+_The name of a namespace must be a valid Cadl identifier._
+
+The `SampleNamespace` can then be used from other places:
+
+```cadl
+model Foo {
+  sample: SampleNamespace.SampleModel;
+}
+```
+
+## Nested namespace
+
+Namespaces can contain sub namespaces providing additional granularity
+
+```cadl
+namespace Foo {
+  namespace Bar {
+    namespace Baz {
+      model SampleModel {}
+    }
+  }
+}
+```
+
+or this can be simplified using `.` notation
+
+```cadl
+namespace Foo.Bar.Baz {
+  model SampleModel {}
+}
+```
+
+The subnamespace can then be used from other places using the fully qualified name.
+
+```cadl
+model A {
+  sample: Foo.Bar.Baz.SampleModel;
+}
+```
+
+## File namespace
+
+A namespace for the file can be provided at the top (After the `import` statements) using a blockless namespace statement
```

Comment:
Github won't let me suggest for some reason, but:

> A namespace for the file

could change to

> A namespace for all declarations contained in a file

to ensure there is no confusion about what is actually being namespaced.